docs/superbench-config.mdx

```diff
@@ -0,0 +1,389 @@
+      prefix: CUDA_VISIBLE_DEVICES={proc_rank}
+      parallel: yes
+  parameters:
+    num_warmup: 100
+    num_steps: 2000000
+    interval: 2000
+```
+
+#### Model-Benchmark
+
+```yaml
+resnet_models:
+  enable: true
+  modes:
+    - name: torch.distributed
+      proc_num: 8
+      node_num: 1
+  frameworks:
+    - pytorch
+  models:
+    - resnet50
+    - resnet101
+    - resnet152
+  parameters:
+    duration: 0
+    num_warmup: 16
+    num_steps: 128
+    batch_size: 128
+    precision:
+      - float32
+      - float16
+    model_action:
+      - train
+```
+
+</TabItem>
+</Tabs>
+
+### `enable`
+
+Enable current benchmark or not, can be overwritten by [`superbench.enable`](#superbenchenable).
+
+* default value: `true`
+
+### `modes`
+
+A list of mode in which the benchmark runs.
+Currently only one mode is supported for each benchmark.
+
+See [`Mode` Schema](#mode-schema) for mode definition.
+
+### `frameworks`
+
+A list of framework in which the benchmark runs.
+Some benchmarks can support multiple frameworks while others only support one.
+
+* accepted values: `[ onnx | pytorch | tf1 | tf2 | none ]`
+* default value: `[ none ]`
+
+### `models`
+
+A list of models to run, only supported in model-benchmark.
+
+* accepted values:
+  ```yaml
+  # pytorch framework
+  [ alexnet | densenet121 | densenet169 | densenet201 | densenet161 | googlenet | inception_v3 |
+    mnasnet0_5 | mnasnet0_75 | mnasnet1_0 | mnasnet1_3 | mobilenet_v2 |
+    resnet18 | resnet34 | resnet50 | resnet101 | resnet152 |
+    resnext50_32x4d | resnext101_32x8d | wide_resnet50_2 | wide_resnet101_2 |
+    shufflenet_v2_x0_5 | shufflenet_v2_x1_0 | shufflenet_v2_x1_5 | shufflenet_v2_x2_0 |
+    squeezenet1_0 | squeezenet1_1 |
+    vgg11 | vgg11_bn | vgg13 | vgg13_bn | vgg16 | vgg16_bn | vgg19_bn | vgg19 |
+    bert-base | bert-large | gpt2-small | gpt2-medium | gpt2-large | gpt2-xl ]
+  ```
+* default value: `[ ]`
+
+### `parameters`
+
+Parameters for benchmark to use, varying for different benchmarks.
+
+## `Mode` Schema
+
+Definition for each benchmark mode, here is an overview of `Mode` configuration structure:
+
+<Tabs
+  defaultValue='schema'
+  values={[
+    {label: 'Schema', value: 'schema'},
+    {label: 'Example', value: 'example'},
+  ]
+}>
+<TabItem value='schema'>
+
+```yaml
+name: enum
+proc_num: int
+node_num: int
+env: dict
+mca: dict
+prefix: str
+parallel: bool
+```
+
+</TabItem>
+<TabItem value='example'>
+
+```yaml
+name: local
+proc_num: 8
+prefix: CUDA_VISIBLE_DEVICES={proc_rank}
+parallel: yes
+```
+
+</TabItem>
+</Tabs>
+
+### `name`
+
+Mode name to use. Here lists available modes:
++ `local`: run benchmark as local process.
++ `torch.distributed`: launch benchmark through [PyTorch DDP](https://pytorch.org/docs/stable/distributed.html#launch-utility), each process will run on one GPU.
++ `mpi`: launch benchmark through MPI, the benchmark implementation could leverage MPI interface.
+
+Some attributes may only be suitable for specific mode.
```

Comment:
Could we have a table for this? Just something like the collective API supported by different backend of pytorch:
https://pytorch.org/docs/stable/distributed.html

Reply:
added